Kernel/Syscalls/resource.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022, Lucas Chollet <lucas.chollet@free.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Process.h>
+
+namespace Kernel {
+
+ErrorOr<FlatPtr> Process::sys$getrusage(int who, Userspace<rusage*> user_usage)
+{
+    VERIFY_PROCESS_BIG_LOCK_ACQUIRED(this)
+
```

Comment:
We should figure out which pledge we should lock this down under, as I don't think we want `getrusage(..)` to be usable by a restricted program.

It seems like we might want to introduce a new pledge just for this. 

@awesomekling Any thoughts?

Reply:
Hi @awesomekling! :^) Can you share your thoughts on this?